### PR TITLE
documentation(CFY-2799, CFY-2800) Topology & network documentation

### PR DESCRIPTION
**Pull request by erezgiga**
Added documentation for blueprint topology & network topology in both blueprint & deployment pages.
Added missing floating node details panel explanation.
Resolves both CFY-2799 & CFY-2800 Jira tasks.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,7 +0,0 @@
-source 'https://rubygems.org'
```
Comment (nir0s) [inline Gemfile]: Why did you delete this? I understand that you needed this for mac... but we can't build without this,

File: source/guide/3.2/webui-blueprints-page.md [hunk 2]
```diff
@@ -13,15 +13,31 @@ terminology_link: reference-terminology.html
 When clicking on the `Blueprints` tab and choosing a blueprint you will be able to choose one of the following:
 
 # Topology
-See the definition [here]({{page.terminology_link}}#topology).<br/>
+The [Topology]({{page.terminology_link}}#topology) is an application’s graph of nodes and their relationships, which describes the lifecycle events or other operations that each node and relationship exposes for use in workflows.<br>
+Each of the blueprint nodes is displayed as a square container, which can contain another nodes. Each of the nodes has a title describing its name, and an icon to indicate the node type.<br>
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: each of the blueprint -> blueprint's
contain another -> contain other
each of the nodes -> each node
the node type -> the node's type

File: source/guide/3.2/webui-blueprints-page.md [hunk 3]
```diff
@@ -13,15 +13,31 @@ terminology_link: reference-terminology.html
...
+Relationships between nodes are marked with arrows, starts from the connected node and ends in the target node.<br>
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: with arrows. A relationship's graphical representation starts from...

File: source/guide/3.2/webui-blueprints-page.md [hunk 4]
```diff
@@ -13,15 +13,31 @@ terminology_link: reference-terminology.html
...
+The topology view shows only the application nodes and not the network nodes. If a node has network dependency, it will be displayed as a bullet icon in the node title.<br>
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: a network dependency

File: source/guide/3.2/webui-blueprints-page.md [hunk 5]
```diff
@@ -13,15 +13,31 @@ terminology_link: reference-terminology.html
...
+The number of node instances is marked in a bullet beside the node type icon.<br>
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: node type -> node's type

File: source/guide/3.2/webui-blueprints-page.md [hunk 6]
```diff
@@ -13,15 +13,31 @@ terminology_link: reference-terminology.html
...
+Pressing a node title will open a side panel with details of the selected node.<br>
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: pressing -> clicking
a node -> a node's

File: source/guide/3.2/webui-blueprints-page.md [hunk 7]
```diff
@@ -13,15 +13,31 @@ terminology_link: reference-terminology.html
...
+
 ![Blueprint topology](/guide/images/ui/ui-blueprint-topology.png)
 
+Pressing a node title will open a side panel with details of the selected node.<br>
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: pressing -> clicking

File: source/guide/3.2/webui-blueprints-page.md [hunk 8]
```diff
@@ -13,15 +13,31 @@ terminology_link: reference-terminology.html
...
+
+![Blueprint node details](/guide/images/ui/ui-blueprint-floating-panel.png)
+
 # Network
-A map of networks topology according to the blueprint topology contained internal and external networks, hosts, routers and ports.<br/>
+A map of networks topology according to the blueprint topology contained internal and external networks, hosts, routers.<br/>
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: networks topology -> network topologies
blueprint topology -> blueprint's topology
contained -> contains

File: source/guide/3.2/webui-blueprints-page.md [hunk 9]
```diff
@@ -13,15 +13,31 @@ terminology_link: reference-terminology.html
...
+The network name is displayed as a grey title, each network contains sub-networks displayed as a colored lines underneath.
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: network -> network's
a colored -> colored

File: source/guide/3.2/webui-blueprints-page.md [hunk 10]
```diff
@@ -13,15 +13,31 @@ terminology_link: reference-terminology.html
...
+Network devices, such as hosts & routers, are displayed as icons, each icon indicates the device type.
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: device -> device's

File: source/guide/3.2/webui-blueprints-page.md [hunk 11]
```diff
@@ -13,15 +13,31 @@ terminology_link: reference-terminology.html
...
+Connections between sub-networks and devices are marked with colored line, by the color of the connected sub-network.<br>
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: colored -> a colored

File: source/guide/3.2/webui-blueprints-page.md [hunk 12]
```diff
@@ -13,15 +13,31 @@ terminology_link: reference-terminology.html
...
+Pressing a device will open a side panel with details of the selected device.<br>
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: pressing -> clicking

File: source/guide/3.2/webui-blueprints-page.md [hunk 13]
```diff
@@ -13,15 +13,31 @@ terminology_link: reference-terminology.html
...
+
 ![Blueprint networks](/guide/images/ui/ui-deployment-networks.jpg)
 
 # Nodes
 A list of nodes according to the blueprint topology.<br/>
+In every node row shown its type, number of instance, and relationships. By pressing the magnifier icon, a side panel will be opened with the selected node details.<br>
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: For every node, its type, number of instance, and relationships are shown. By clicking the magnifier icon, a side panel will be opened with the selected node's details.

File: source/guide/3.2/webui-blueprints-page.md [hunk 14]
```diff
@@ -13,15 +13,31 @@ terminology_link: reference-terminology.html
 When clicking on the `Blueprints` tab and choosing a blueprint you will be able to choose one of the following:
 
 # Topology
-See the definition [here]({{page.terminology_link}}#topology).<br/>
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: I see that this description is pretty much cloned in the blueprints page and deployments page. Maybe we can deduplicate somehow?

File: source/guide/3.2/webui-deployments-page.md [hunk 15]
```diff
@@ -13,15 +13,36 @@ terminology_link: reference-terminology.html
 When clicking on the `Deployments` tab and choosing a deployment you will be able to choose one of the following:
 
 # Topology
-See the definition [here]({{page.terminology_link}}#topology).<br/>
+The [Topology]({{page.terminology_link}}#topology) is an application’s graph of nodes and their relationships, which describes the lifecycle events or other operations that each node and relationship exposes for use in workflows.<br>
```
Comment (nir0s) [inline source/guide/3.2/webui-deployments-page.md]: provide the same fixes as above.

File: Gemfile (hunk 1 above)
Comment (erezgiga) [inline Gemfile]: Sorry, my mistake, I had to delete it in order to run Jekyll over OSX, but it shouldn't commit this change. Will revert this deletion.

File: source/guide/3.2/webui-blueprints-page.md (hunk 14 above)
Comment (erezgiga) [inline source/guide/3.2/webui-blueprints-page.md]: Most of the views in both blueprints & deployments are pretty much the same, but there are a few small differences. In deployment views we show the runtime properties instead of the static properties in the blueprint views.

File: source/guide/3.2/webui-blueprints-page.md [hunk 16]
```diff
@@ -7,21 +7,38 @@ abstract: Blueprint Page Reference
...
+relationships_link: dsl-spec-relationships.html
 ---
 {%summary%}This is a view of a single blueprint in Cloudify's User Interface{%endsummary%}.
 
 When clicking on the `Blueprints` tab and choosing a blueprint you will be able to choose one of the following:
 
 # Topology
-See the definition [here]({{page.terminology_link}}#topology).<br/>
+The [Topology]({{page.terminology_link}}#topology) is an application’s graph of nodes and their relationships, which describes the lifecycle events or other operations that each node and relationship exposes for use in workflows.<br>
+Each of the blueprint nodes is displayed as a square container, which can contain another nodes. Each of the nodes has a title describing its name, and an icon to indicate the [node type]({{page.node_types_link}}).<br>
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: blueprint nodes -> blueprint's nodes
another nodes -> other nodes...

I see some of these around, which I already commented on. please fix.

File: source/guide/3.2/webui-blueprints-page.md [hunk 17]
```diff
@@ -7,22 +7,39 @@ abstract: Blueprint Page Reference
...
+relationships_link: dsl-spec-relationships.html
 ---
 {%summary%}This is a view of a single blueprint in Cloudify's User Interface{%endsummary%}.
 
 When clicking on the `Blueprints` tab and choosing a blueprint you will be able to choose one of the following:
 
 # Topology
-See the definition [here]({{page.terminology_link}}#topology).<br/>
+The [Topology]({{page.terminology_link}}#topology) is an application’s graph of nodes and their relationships, which describes the lifecycle events or other operations that each node and relationship exposes for use in workflows.<br>
+Each of the blueprint's nodes is displayed as a square container, which can contain other nodes. Each of the nodes has a title describing its name, and an icon to indicate the [node type]({{page.node_types_link}}).<br>
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: Each of the nodes > each node
node type > node's type

File: source/guide/3.2/webui-blueprints-page.md [hunk 18]
```diff
@@ -7,22 +7,39 @@ abstract: Blueprint Page Reference
...
+[Relationships]({{page.relationships_link}}) between nodes are marked with arrows, starts from the connected node and ends in the target node.<br>
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: ends in > ends at

File: source/guide/3.2/webui-blueprints-page.md [hunk 19]
```diff
@@ -7,22 +7,39 @@ abstract: Blueprint Page Reference
...
+The topology view shows only the application nodes and not the network nodes. If a node has network dependency, it will be displayed as a bullet icon in the node title.<br>
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: a network dependency

File: source/guide/3.2/webui-blueprints-page.md [hunk 20]
```diff
@@ -7,22 +7,39 @@ abstract: Blueprint Page Reference
...
+The number of [node instances]({{page.terminology_link}}#node-instance) is marked in a bullet beside the node type icon.<br>
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: node's type icon...

I already commented on most of these.. please review and fix.

File: source/guide/3.2/webui-blueprints-page.md [hunk 21]
```diff
@@ -7,22 +7,39 @@ abstract: Blueprint Page Reference
...
+Network devices, such as hosts & routers, are displayed as icons, each icon indicates the device's type.
+Connections between sub-networks and devices are marked with a colored line, by the color of the connected sub-network.<br>
+Clicking a device will open a side panel with details of the selected device.<br>
+
 ![Blueprint networks](/guide/images/ui/ui-deployment-networks.jpg)
 
 # Nodes
 A list of nodes according to the blueprint topology.<br/>
-![Blueprint nodes](/guide/images/ui/ui-deployment-nodes.jpg)
+For every node, its type, number of instance, and relationships are shown. By clicking the magnifier icon, a side panel will be opened with the selected node's details.
```
Comment (nir0s) [inline source/guide/3.2/webui-blueprints-page.md]: number of instances

File: source/guide/3.2/webui-deployments-page.md (hunk 15 above)
Comment (erezgiga) [inline source/guide/3.2/webui-deployments-page.md]: Done.

**Comment by m1keil** [comment]
LGTM except for the images that are placed under /images and contain no version in file name or path. might be a problem moving forward.

**Comment by erezgiga** [comment]
@m1keil None of the documentation images (dsl, plugins etc.) contains the version number. I don't see any reason that the UI should. I understand the problem, but this should be done for all documentation images and not just UI.

**Comment by m1keil** [comment]
it's just that UI photos are much more likely to change in next version...
